### PR TITLE
Add blog post for release of hpy-0.0.3

### PR DESCRIPTION
**Pull request by fangerer**
As we've been talking in IRC, I prepared a short blog post about the release of 0.0.3.
I took the post for 0.0.2 as template.
Please tell me if you think something is missing or wrong.

### CODE REVIEW

File: posts/2021/10/hpy-0.0.3.md
Comment (antocuni) [inline posts/2021/10/hpy-0.0.3.md:101]: This sentence is a bit obscure. What about this?

```suggestion
In HPy, direct comparisons of handles is forbidden by design, because distinct handles can point to the same object. To compare objects by identity, you need to use `HPy_Is`:
```

File: posts/2021/10/hpy-0.0.3.md
Comment (antocuni) [inline posts/2021/10/hpy-0.0.3.md:156]: is it actually true? I think that at the moment, a double-close doesn't crash, see https://github.com/hpyproject/hpy/issues/258#issuecomment-958785897

The crash happens if you try to actually use a closed handle.
Note that I am more and more convinced that allowing multiple `HPy_Close` was a mistake and that we should change the current behavior.

File: posts/2021/10/hpy-0.0.3.md
Comment (antocuni) [inline posts/2021/10/hpy-0.0.3.md:172]: I agree with @fangerer and @rlamy : I introduced `set_on_invalid_handle` mostly for our internal tests and we should not advertise it.
I think the only reasonable thing to do is to abort as soon as we detect an invalid usage of a closed error, because it's basically equivalent to a segfault: we no longer know whether the underlying `PyObject *` points to valid memory, so the program might crash soon anyway and/or do unpredictable things.

Our tests are allowed to use it because they are written in such a way to ensure that the memory is still valid, but that's just a property of our tests

**Review by antocuni — CHANGES_REQUESTED** [review]
Thank you for the blog post.
Looks overall good, but see my inline comments

File: posts/2021/10/hpy-0.0.3.md
Comment (fangerer) [reply posts/2021/10/hpy-0.0.3.md:172]: Okay, I'll remove the description for `set_on_invalid_handle` and will fix the above example.

**Comment by fangerer** [comment]
> Maybe we worth waiting with publishing this until the docs are fixed in [hpyproject/hpy#253](https://github.com/hpyproject/hpy/pull/253)

@steve-s I think this PR does not have a lot to do with your PR since this blog post is about 0.0.3 and we won't backport your changes to 0.0.3, right?

**Comment by steve-s** [comment]
> we won't backport your changes to 0.0.3, right?

Ah yes, it would have to be back-ported. The unfortunate thing is that the docs show wrong examples. If the release blog post gets some traction and brings people over to the documentation website, it would be good to show working examples there, but the blog post itself links to the "latest" tag of the documentation so as long as it's merged to master...

**Comment by fangerer** [comment]
Hmm, right. Okay, I can wait until your PR is merged. I don't mind.

File: posts/2021/10/hpy-0.0.3.md
Comment (fangerer) [reply posts/2021/10/hpy-0.0.3.md]: I've removed this example since it does no longer make sense.

**Comment by fangerer** [comment]
@antocuni I addressed your comments (remove the invalid handle hook advertisement). Please have another look.

**Review by antocuni — APPROVED** [review]
Thank you for addressing my comments and sorry for the delay. LGTM

**Pull request merged**
